Sources/AsyncProcess/ProcessExecutor.swift

```diff
@@ -424,10 +532,12 @@ public final actor ProcessExecutor {
       )
       precondition(worked, "illegal running state \(original)")
 
-      if p.terminationReason == .uncaughtSignal {
-        terminationStreamProducer.yield(.signal(p.terminationStatus))
-      } else {
-        terminationStreamProducer.yield(.exit(p.terminationStatus))
+      for _ in 0..<2 {
```

Comment:
Could you elaborate on why we need to check the reason twice?

Reply:
Yes, two tasks are waiting for the exit status:

1. the actual user
2. the termination sequence stuff that stops itself as soon as the process exits

Reply:
Ah, got it. `waitForChildToExit` is now awaited twice.

Reply:
no